app/src/main/java/org/simple/clinic/summary/bloodsugar/BloodSugarSummaryViewEffectHandler.kt

```diff
@@ -2,25 +2,36 @@ package org.simple.clinic.summary.bloodsugar
 
 import com.spotify.mobius.rx2.RxMobius
 import io.reactivex.ObservableTransformer
+import io.reactivex.Scheduler
 import org.simple.clinic.bloodsugar.BloodSugarRepository
 import org.simple.clinic.util.scheduler.SchedulersProvider
 
 object BloodSugarSummaryViewEffectHandler {
 
   fun create(
       bloodSugarRepository: BloodSugarRepository,
-      schedulersProvider: SchedulersProvider
+      schedulersProvider: SchedulersProvider,
+      uiActions: UiActions
   ): ObservableTransformer<BloodSugarSummaryViewEffect, BloodSugarSummaryViewEvent> {
     return RxMobius
         .subtypeEffectHandler<BloodSugarSummaryViewEffect, BloodSugarSummaryViewEvent>()
-        .addTransformer(FetchBloodSugarSummary::class.java) { effect ->
-          effect
-              .flatMap {
-                bloodSugarRepository.latestMeasurements(it.patientUuid, 100)
-                    .subscribeOn(schedulersProvider.io())
-              }
-              .map { BloodSugarSummaryFetched(it) }
-        }
+        .addTransformer(FetchBloodSugarSummary::class.java, fetchBloodSugarMeasurements(bloodSugarRepository, schedulersProvider.ui()))
```

Comment:
This commit is doing two things:

- Extracting this transformer into a separate method, and
- Adding an action to open the sheet

It is not necessary to change this, but we should try to restrict each commit to only one change.

Reply:
I am not adding the transformer in this commit. I am just refactoring the code to a separate method since there are now more than one effects being handled.

Reply:
Yeah, the transformer was added in a previous commit. What I meant was in this commit, we are both extracting the method as well as adding another action.

It's not necessary to split this commit right now. Just a comment for future PRs.